### PR TITLE
Add pyenv support

### PR DESCRIPTION
**Pull request by hazedic**
### How Has This Been Tested? ###

| Architecture | Yes/No                   | Comments               |
|--------------|:------------------------:|------------------------|
| x86-32       | :heavy_multiplication_x: |    |
| x86-64       | :heavy_check_mark: |                        |
| ARM          | :heavy_multiplication_x: |                        |
| AARCH64      | :heavy_multiplication_x: |                        |
| MIPS         | :heavy_multiplication_x: |                        |
| POWERPC      | :heavy_multiplication_x: |                        |
| SPARC        | :heavy_multiplication_x: |                        |

### Checklist ###

- [x] My code follows the code style of this project.
- [x] My change includes a change to the documentation, if required.
- [x] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -9146,6 +9146,14 @@ def __gef_prompt__(current_prompt):
             "Consider updating to GDB {} or higher.".format(".".join(GDB_MIN_VERSION)))
 
     else:
+        import commands
+        import site
+        if commands.getstatusoutput('pyenv root'):
```
Comment (Grazfather) [inline gef.py]: Why not use subrocess?

**Comment by Grazfather** [comment]
This looks interesting. `site` is a nicer way to edit `sys.path`, basically?

Could you please explain it a bit better? If you look at our circle config we manually set PYTHONPATH. We could maybe use this instead.

Use case examples would be great.

File: gef.py (hunk 1 above)
Comment (hazedic) [reply gef.py]: screen shot 2018-10-13 at 3 03 02 am

I changed to use subprocess.

**Comment by hazedic** [comment]
I frequently use to switch between multiple versions of Python.

So I have added the code to automatically find the version of the site-packages directory whenever the version changes.

File: gef.py [hunk 2]
```diff
@@ -9146,6 +9147,12 @@ def __gef_prompt__(current_prompt):
             "Consider updating to GDB {} or higher.".format(".".join(GDB_MIN_VERSION)))
 
     else:
+        PYENV_ROOT = subprocess.check_output(["pyenv", "root"]).strip()
```
Comment (Grazfather) [inline gef.py]: This will crash if pyenv is not installed

File: gef.py [hunk 3]
```diff
@@ -9146,6 +9147,12 @@ def __gef_prompt__(current_prompt):
...
+        if PYENV_ROOT:
+            PYENV_VERSION = subprocess.check_output(["pyenv" "version-name"]).strip()
```
Comment (Grazfather) [inline gef.py]: Missing command between args. Did you test this?

**Comment by hazedic** [comment]
I tested on macOS High Sierra, Ubuntu 16.04, 18.04.

**Comment by hazedic** [comment]
Please suggest a better way to support pyenv (Python version management).

**Comment by Grazfather** [comment]
use our `which()` function to test for whether there is `pyenv` on the system.

File: gef.py [hunk 4]
```diff
@@ -9148,11 +9148,12 @@ def __gef_prompt__(current_prompt):
 
     else:
         try:
+            pyenv = which("pyenv")
             PYENV_ROOT = subprocess.check_output(["pyenv", "root"]).strip()
```
Comment (hugsy) [inline gef.py]: Replace with 
```
PYENV_ROOT = subprocess.check_output([pyenv, "root"]).strip()
```

File: gef.py [hunk 5]
```diff
@@ -9148,11 +9148,12 @@ def __gef_prompt__(current_prompt):
...
             PYENV_VERSION = subprocess.check_output(["pyenv", "version-name"]).strip()
```
Comment (hugsy) [inline gef.py]: Same

**Comment by hugsy** [comment]
Hey @hazedic, 

Thanks for the PR. Can you make those final change this week ? We'd like to merge your PR for the next release of GEF coming up by next weekend.

Cheers,

File: gef.py [hunk 6]
```diff
@@ -9146,6 +9147,15 @@ def __gef_prompt__(current_prompt):
             "Consider updating to GDB {} or higher.".format(".".join(GDB_MIN_VERSION)))
 
     else:
+        try:
+            pyenv = which("pyenv")
+            PYENV_ROOT = subprocess.check_output(["pyenv", "root"]).strip()
+            PYENV_VERSION = subprocess.check_output(["pyenv", "version-name"]).strip()
+            site_packages_dir = os.path.join(PYENV_ROOT, "versions", PYENV_VERSION, "lib", "python{}".format(PYENV_VERSION[:3]), "site-packages")
```
Comment (Grazfather) [inline gef.py]: Split this line up pllease

**Comment by hazedic** [comment]
I applied the requirements. check it plz. 
Thanks.

**Comment by hugsy** [comment]
> I applied the requirements. check it plz.
> Thanks.

@hazedic 

Please check out @Grazfather's latest review and adjust your code so we can merge.

Thanks

File: gef.py (hunk 6 above)
Comment (Grazfather) [reply gef.py]:
```suggestion
            site_packages_dir = os.path.join(PYENV_ROOT, "versions", PYENV_VERSION, "lib",
                                             "python{}".format(PYENV_VERSION[:3]), "site-packages")
```

Comment (hazedic) [reply gef.py]: I got it.

**Comment by hugsy** [comment]
:+1:

**Pull request merged**